include/amici_model.h

```diff
@@ -300,30 +312,282 @@ namespace amici {
+         */
+        std::vector<int> getParameterList() const {
+            return plist_;
+        }
+
+        /**
+         * @brief setParameterList
+         * @param plist
+         */
+        void setParameterList(std::vector<int> const& plist) {
+            for(auto const& idx: plist)
+                if(idx < 0 || idx >= np())
+                    throw AmiException("Indices in plist must be in [0..np]");
+            this->plist_ = plist;
+
+            initializeVectors();
+        }
+
+        /**
+         * @brief getInitialStates
+         * @return
+         */
+        std::vector<realtype> getInitialStates() const {
+            return x0data;
+        }
+
+        /**
+         * @brief setInitialStates
+         * @param x0
+         */
+        void setInitialStates(std::vector<realtype> const& x0) {
+            if(x0.size() != (unsigned) nx)
+                throw AmiException("Dimension mismatch.");
+            this->x0data = x0;
+        }
+
+        /**
+         * @brief getInitialStateSensitivities
+         * @return
+         */
+        std::vector<realtype> getInitialStateSensitivities() const {
+            return sx0data;
+        }
+
+        /**
+         * @brief setInitialStateSensitivities
+         * @param sx0
+         */
+        void setInitialStateSensitivities(std::vector<realtype> const& sx0) {
+            if(sx0.size() != (unsigned) nx * nplist())
+                throw AmiException("Dimension mismatch.");
+            this->sx0data = sx0;
+        }
+
+        /**
+         * @brief getParameterScaling
+         * @return
+         */
+        std::vector<realtype> getParameterScaling() const {
+            return pbar;
+        }
+
+        /**
+         * @brief setParameterScaling
+         * @param pbar
+         */
+        void setParameterScaling(std::vector<realtype> const& pbar) {
+            if(pbar.size() != (unsigned) nplist())
+                throw AmiException("Dimension mismatch.");
+            this->pbar = pbar;
+        }
+
+        /** initial timepoint
+         *  @return timepoint
+         */
+        double t0() const{
+            return tstart;
+        }
+
+        /**
+         * @brief setT0
+         * @param t0
+         */
+        void setT0(double t0) {
+            tstart = t0;
+        }
+
+        /** entry in parameter list
+          * @param pos index
+          * @return entry
+          */
+        const int plist(int pos) const{
+            return plist_.at(pos);
+        }
+
+        /**
+         * @brief unscaleParameters
+         * @param bufferUnscaled
+         */
+        void unscaleParameters(double *bufferUnscaled) const;
+
+        /**
+         * @brief Require computation of sensitivities for all parameters p [0..np[
+         * in natural order.
+         */
+        void requireSensitivitiesForAllParameters() {
+            plist_.resize(np());
+            std::iota(plist_.begin(), plist_.end(), 0);
+        }
+
```

Comment:
do we want unit tests for these routines?

Reply:
For all the setters? As soon as there is proper input checking, yes.